kyaml/yaml/types.go

```diff
@@ -722,22 +722,36 @@ func (rn *RNode) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
-// ConvertJSONToYamlNode parses input json string and returns equivalent yaml node
-func ConvertJSONToYamlNode(jsonStr string) (*RNode, error) {
-	var body map[string]interface{}
-	err := json.Unmarshal([]byte(jsonStr), &body)
+// ConvertJSONToYAMLNode parses input json string and returns equivalent yaml node
+func ConvertJSONToYAMLNode(jsonStr string) (*RNode, error) {
+	yml, err := ConvertJSONToYAMLString(jsonStr)
 	if err != nil {
 		return nil, err
 	}
-	yml, err := yaml.Marshal(body)
+	node, err := Parse(yml)
 	if err != nil {
 		return nil, err
 	}
-	node, err := Parse(string(yml))
+	return node, nil
+}
+
+// ConvertJSONToYAMLString parses input json string and returns equivalent yaml string
+func ConvertJSONToYAMLString(jsonStr string) (string, error) {
+	var body map[string]interface{}
+	err := json.Unmarshal([]byte(jsonStr), &body)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	return node, nil
+	yml, err := yaml.Marshal(body)
+	if err != nil {
+		return "", err
+	}
+	return string(yml), nil
+}
+
+func ValidJSON(s string) bool {
+	var js map[string]interface{}
```

Comment:
Why not?

```go
func IsJSON(str string) bool {
    var js json.RawMessage
    return json.Unmarshal([]byte(str), &js) == nil
}
```

https://stackoverflow.com/questions/22128282/how-to-check-string-is-in-json-format

Reply:
Done.